src/utils/nlpUtils/OPLparser.py

```diff
@@ -16,113 +16,205 @@
 import spacy
 # Internal Import
 
-def OPLentityParser(filename):
-  '''
-  This method extracts all the form and function entities out of the OPL html file and it puts them in two separate lists
-  @in: filename, file name containing the OPL text
-  @out: objectList, list, list of form elements contained in the OPL file
-  @out: processList, list, list of function elements contained in the OPL file
-  '''
-  objectList = []
-  processList = []
-  with open(filename) as fp:
-    soup = BeautifulSoup(fp, "html.parser")
-    elements = soup.find("font").findChildren()
-    for element in elements:
-      if element.has_attr('color'):
-        if element['color'] == '#006d00':
+class OPMobject(object):
+  def __init__(self, filename):
+    self.filename = filename
+    self.objectList  = []
+    self.processList = []
+    self.edge_colors = []
+    self.node_colors = []
+    self.sentences   = None
+    self.opmGraph    = None
+    self.links2OPMs  = []
+    self.acronyms    = {}
+
+    self.OPLentityParser()
+    self.OPLtextParser()
+    self.OPLparser()
+
+    
+  def OPLentityParser(self):
+    '''
+    This method extracts all the form and function entities out of the OPL html file and it puts them in two separate lists
+    @in: filename, file name containing the OPL text
+    @out: objectList, list, list of form elements contained in the OPL file
+    @out: processList, list, list of function elements contained in the OPL file
```

Comment:
Please update the docstring, it seems this function accept no input argument, and return no output.